lang/py/src/avro/io.py

```diff
@@ -341,6 +402,59 @@ def write_double(self, datum):
     self.write(chr((bits >> 48) & 0xFF))
     self.write(chr((bits >> 56) & 0xFF))
 
+  def write_decimal_bytes(self, datum, scale):
+    """
+    Decimal in bytes are encoded as long. Since size of packed value in bytes for
+    signed long is 8, 8 bytes are written.
+    """
+    if isinstance(datum, Decimal):
+      datum = datum.to_eng_string()
+    else:
+      datum = str(datum)
```

Comment:
Using a number's string representation to infer the precision is dangerous. It depends on python's internal number to string conversion and can easily lead to incorrect or inconsistent results. For example:

+ `str(0.1234567890123) == str(0.1234567890124)`, so this (silently) truncates precision at 12.
+ `str(sum(0.1 for _ in range(1000000))) != '100000.0'`, so how a number is computed will affect the success of this function (which would be fine if it was only precision loss; but in this case, one number would cause the write to fail while the other would succeed).

Accepting raw numbers is convenient, but shouldn't come at the cost of corrupting data. It would be safer to only accept `Decimal`s (at least by default). Also, these string conversions are likely inefficient.